examples/02_decoding/plot_oasis_vbm.py

```diff
@@ -39,132 +45,148 @@
 #          Virgile Fritsch, <virgile.fritsch@inria.fr>, Apr 2014
 #          Gael Varoquaux, Apr 2014
+#          Andres Hoyos-Idrobo, Apr 2017
+
+######################################################################
+# Loading the data
+######################################################################
+# First, we load the data
 import numpy as np
-import matplotlib.pyplot as plt
 from nilearn import datasets
+n_subjects = 200  # increase this number if you have more RAM on your box
+dataset_files = datasets.fetch_oasis_vbm(n_subjects=n_subjects)
+age = dataset_files.ext_vars['age'].astype(float)
+gm_imgs = dataset_files.gray_matter_maps
+
+# Split data into training set and test set
+from sklearn.model_selection import train_test_split
+gm_imgs_train, gm_imgs_test, age_train, age_test = train_test_split(
+    gm_imgs, age, train_size=.6, random_state=0)
+######################################################################
+# Preprocess the mask
+# --------------------
+
+# First we use the :class:`nilearn.input_data.NiftiMasker` to extract 
+# fMRI data on a mask and convert it to data series.
 from nilearn.input_data import NiftiMasker
+nifti_masker = NiftiMasker(standardize=False, smoothing_fwhm=2,
+                           memory='nilearn_cache')
+gm_maps_masked = nifti_masker.fit_transform(gm_imgs_train)
 
-n_subjects = 100  # more subjects requires more memory
-
-############################################################################
-# Load Oasis dataset
-# -------------------
-oasis_dataset = datasets.fetch_oasis_vbm(n_subjects=n_subjects)
-gray_matter_map_filenames = oasis_dataset.gray_matter_maps
-age = oasis_dataset.ext_vars['age'].astype(float)
-
-# print basic information on the dataset
-print('First gray-matter anatomy image (3D) is located at: %s' %
-      oasis_dataset.gray_matter_maps[0])  # 3D data
-print('First white-matter anatomy image (3D) is located at: %s' %
-      oasis_dataset.white_matter_maps[0])  # 3D data
-
-#############################################################################
-# Preprocess data
-# ----------------
-nifti_masker = NiftiMasker(
-    standardize=False,
-    smoothing_fwhm=2,
-    memory='nilearn_cache')  # cache options
-gm_maps_masked = nifti_masker.fit_transform(gray_matter_map_filenames)
-n_samples, n_features = gm_maps_masked.shape
-print("%d samples, %d features" % (n_subjects, n_features))
-
-############################################################################
-# Prediction pipeline with ANOVA and SVR
-# ---------------------------------------
-print("ANOVA + SVR")
-# Define the prediction function to be used.
-# Here we use a Support Vector Classification, with a linear kernel
-from sklearn.svm import SVR
-svr = SVR(kernel='linear')
-
-# Dimension reduction
-from sklearn.feature_selection import VarianceThreshold, SelectKBest, \
-        f_regression
-
-# Remove features with too low between-subject variance
```

Comment:
You need a block separation here:
"
##########################################################################
"